### PR TITLE
Wrap API router in recoveryMiddleware 

### LINKED ISSUE #6105: [BUG] hitting nonexistent API endpoint under /v1/harvester/ gives 500 error and panic backtrace in log
**Describe the bug**
Hit a nonexistent API endpoint directly under `/v1/harvester/` (for example `https://192.168.0.131/v1/harvester/wtf`), and the client will get a 500 internal server error, and the text "EOF" in the response body. Meanwhile, in the logs of the harvester pod, we see:

```
2024/07/02 09:17:48 http: panic serving pipe: runtime error: invalid memory address or nil pointer dereference
goroutine 7890 [running]:
net/http.(*conn).serve.func1()
	/usr/local/go/src/net/http/server.go:1868 +0xb9
panic({0x26f2760?, 0x4ef22f0?})
	/usr/local/go/src/runtime/panic.go:920 +0x270
github.com/harvester/harvester/vendor/github.com/rancher/apiserver/pkg/server.(*Server).handle(0xc0008e8930, 0xc000ae7e00, 0x100?)
	/go/src/github.com/harvester/harvester/vendor/github.com/rancher/apiserver/pkg/server/server.go:154 +0x814
github.com/harvester/harvester/vendor/github.com/rancher/apiserver/pkg/server.(*Server).Handle(0xc000010408?, 0x32400d8?)
	/go/src/github.com/harvester/harvester/vendor/github.com/rancher/apiserver/pkg/server/server.go:101 +0x17
github.com/harvester/harvester/vendor/github.com/rancher/steve/pkg/server/handler.New.(*apiServer).apiHandler.func3({0x32400d8?, 0xc007c9e460?}, 0xc008bd5580?)
	/go/src/github.com/harvester/harvester/vendor/github.com/rancher/steve/pkg/server/handler/apiserver.go:98 +0x7e
net/http.HandlerFunc.ServeHTTP(0xc008bede00?, {0x32400d8?, 0xc007c9e460?}, 0xc008bd5580?)
	/usr/local/go/src/net/http/server.go:2136 +0x29
github.com/harvester/harvester/vendor/github.com/rancher/steve/pkg/auth.init.ToMiddleware.func1.1({0x32400d8, 0xc007c9e460}, 0x10?)
	/go/src/github.com/harvester/harvester/vendor/github.com/rancher/steve/pkg/auth/filter.go:172 +0x24c
net/http.HandlerFunc.ServeHTTP(0xc008e4ebd0?, {0x32400d8?, 0xc007c9e460?}, 0x2b13080?)
	/usr/local/go/src/net/http/server.go:2136 +0x29
github.com/harvester/harvester/vendor/github.com/rancher/apiserver/pkg/urlbuilder.RedirectRewrite.func1({0x32400d8?, 0xc007c9e460}, 0xc008bede00)
	/go/src/github.com/harvester/harvester/vendor/github.com/rancher/apiserver/pkg/urlbuilder/redirect.go:14 +0x152
net/http.HandlerFunc.ServeHTTP(0xc008bedd00?, {0x32400d8?, 0xc007c9e460?}, 0xc009bdb420?)
	/usr/local/go/src/net/http/server.go:2136 +0x29
github.com/harvester/harvester/vendor/github.com/gorilla/mux.(*Router).ServeHTTP(0xc0009fe900, {0x32400d8, 0xc007c9e460}, 0xc008bedc00)
	/go/src/github.com/harvester/harvester/vendor/github.com/gorilla/mux/mux.go:212 +0x1c5
github.com/harvester/harvester/vendor/github.com/rancher/steve/pkg/aggregation.ListenAndServe.ListenAndServe.ToMiddleware.func1.func2({0x32400d8, 0xc007c9e460}, 0x0?)
	/go/src/github.com/harvester/harvester/vendor/github.com/rancher/steve/pkg/auth/filter.go:172 +0x24f
net/http.HandlerFunc.ServeHTTP(0x4f78f20?, {0x32400d8?, 0xc007c9e460?}, 0xc00aaa1b50?)
	/usr/local/go/src/net/http/server.go:2136 +0x29
net/http.serverHandler.ServeHTTP({0xc008e4eba0?}, {0x32400d8?, 0xc007c9e460?}, 0x6?)
	/usr/local/go/src/net/http/server.go:2938 +0x8e
net/http.(*conn).serve(0xc008e5a750, {0x32514a0, 0xc002605650})
	/usr/local/go/src/net/http/server.go:2009 +0x5f4
created by net/http.(*Server).Serve in goroutine 1225
	/usr/local/go/src/net/http/server.go:3086 +0x5cb
```

Note that the harvester pod keeps running just fine - `net/http` appears to have caught and logged the panic, so this isn't critical, it's just weird and ugly.

**To Reproduce**
1. Go to https://192.168.0.131/v1/harvester/wtf (or whatever your management IP address is)
2. Observe that your browser sees a 500 internal server error, and the text "EOF".

**Expected behavior**
A 404 not found error, and no panic/backtrace in the logs.

**Support bundle**
N/A

**Environment**
 - Harvester ISO version: master as at 2024-07-02
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): virtualized

**Additional context**
Based on the backtrace, the line that's triggering the panic is:

https://github.com/harvester/harvester/blob/6b79086f42d005c19070b760a877fe27ccb4767a/vendor/github.com/rancher/apiserver/pkg/server/server.go#L154

Given the URL is a path to something that doesn't exist, my money is on `apiOp.Schema` being nil at this point, and that trying to get `apiOp.Schema.ID` is what's causing the panic, which technically makes this a bug in... wait a minute! ~I've just found a PR that will fix this: https://github.com/rancher/apiserver/pull/13~ This should do it: https://github.com/rancher/apiserver/commit/8ad6ac8d69cd5b0605340047c7ea9e675544aac2#diff-104a11712684e520f0affa95e6a053faeb713306a10f69ee99b9729cc9dd1996R155

### PR DESCRIPTION
**Pull request by tserong**
This ensures that if any panic occurs when an API endpoint is hit, that the sever will return a 500 error to the client, and log the panic message and backtrace to the harvester pod logs.  It's possibly not strictly necessary that we do this, given that in the only panic I've actually seen (see #6105), net/http already logs the backtrace, but, this makes a nice catchall in case there's anything net/http could miss.

Also, this particular change has an interesting side effect.  The panic I'm referring to above is when you hit a nonexistent API endpoint like https://192.168.0.131/v1/harvester/wtf.  What I really want to see in that case is a 404 not found error.  It turns out that in that specific case, we've already called WriteHeader() to set a 404 response code, so the subsequent 500 set by this recovery handler does nothing, but we still get the panic in the logs.  So this (kinda) fixes #6105, at least from the client's perspective ;-)

~Note: no actual error message is returned to the client during a panic.  It's just a 500 HTTP response code with no body - you have to check the pod logs if you want to find out what went wrong.  Here's an example of the logs:~

```
2024/07/02 09:36:01 http: superfluous response.WriteHeader call from github.com/harvester/harvester/vendor/github.com/gorilla/handlers.recoveryHandler.ServeHTTP.func1 (recovery.go:73)
2024/07/02 09:36:01 runtime error: invalid memory address or nil pointer dereference
2024/07/02 09:36:01 goroutine 8540 [running]:
runtime/debug.Stack()
	/usr/local/go/src/runtime/debug/stack.go:24 +0x5e
github.com/harvester/harvester/vendor/github.com/gorilla/handlers.recoveryHandler.log({{0x3260fa0, 0xc0009c7a40}, {0x0, 0x0}, 0x1}, {0xc00d3700f0?, 0x0?, 0x0?})
	/go/src/github.com/harvester/harvester/vendor/github.com/gorilla/handlers/recovery.go:89 +0xb8
github.com/harvester/harvester/vendor/github.com/gorilla/handlers.recoveryHandler.ServeHTTP.func1()
	/go/src/github.com/harvester/harvester/vendor/github.com/gorilla/handlers/recovery.go:74 +0xbe
panic({0x272b240?, 0x4f7ae80?})
	/usr/local/go/src/runtime/panic.go:914 +0x21f
github.com/harvester/harvester/vendor/github.com/rancher/apiserver/pkg/server.(*Server).handle(0xc005eed500, 0xc0060d2690, 0x100?)
	/go/src/github.com/harvester/harvester/vendor/github.com/rancher/apiserver/pkg/server/server.go:154 +0x814
github.com/harvester/harvester/vendor/github.com/rancher/apiserver/pkg/server.(*Server).Handle(0xc003859dd0?, 0x327b840?)
	/go/src/github.com/harvester/harvester/vendor/github.com/rancher/apiserver/pkg/server/server.go:101 +0x17
github.com/harvester/harvester/vendor/github.com/rancher/steve/pkg/server/handler.New.(*apiServer).apiHandler.func3({0x327b840?, 0xc006cf40e0?}, 0xc003080840?)
	/go/src/github.com/harvester/harvester/vendor/github.com/rancher/steve/pkg/server/handler/apiserver.go:98 +0x7e
net/http.HandlerFunc.ServeHTTP(0xc00d930300?, {0x327b840?, 0xc006cf40e0?}, 0xc003080840?)
	/usr/local/go/src/net/http/server.go:2136 +0x29
github.com/harvester/harvester/vendor/github.com/rancher/steve/pkg/auth.init.ToMiddleware.func1.1({0x327b840, 0xc006cf40e0}, 0x10?)
	/go/src/github.com/harvester/harvester/vendor/github.com/rancher/steve/pkg/auth/filter.go:172 +0x24c
net/http.HandlerFunc.ServeHTTP(0xc007ca7d40?, {0x327b840?, 0xc006cf40e0?}, 0x2b4c360?)
	/usr/local/go/src/net/http/server.go:2136 +0x29
github.com/harvester/harvester/vendor/github.com/rancher/apiserver/pkg/urlbuilder.RedirectRewrite.func1({0x327b840?, 0xc006cf40e0}, 0xc00d930300)
	/go/src/github.com/harvester/harvester/vendor/github.com/rancher/apiserver/pkg/urlbuilder/redirect.go:14 +0x152
net/http.HandlerFunc.ServeHTTP(0xc00d930200?, {0x327b840?, 0xc006cf40e0?}, 0x418b28?)
	/usr/local/go/src/net/http/server.go:2136 +0x29
github.com/harvester/harvester/vendor/github.com/gorilla/mux.(*Router).ServeHTTP(0xc0009c7a40, {0x327b840, 0xc006cf40e0}, 0xc009c7bf00)
	/go/src/github.com/harvester/harvester/vendor/github.com/gorilla/mux/mux.go:210 +0x1c5
github.com/harvester/harvester/vendor/github.com/gorilla/handlers.recoveryHandler.ServeHTTP({{0x3260fa0, 0xc0009c7a40}, {0x0, 0x0}, 0x1}, {0x327b840?, 0xc006cf40e0?}, 0x0?)
	/go/src/github.com/harvester/harvester/vendor/github.com/gorilla/handlers/recovery.go:78 +0xd9
github.com/harvester/harvester/vendor/github.com/rancher/steve/pkg/aggregation.ListenAndServe.ListenAndServe.ToMiddleware.func1.func2({0x327b840, 0xc006cf40e0}, 0x0?)
	/go/src/github.com/harvester/harvester/vendor/github.com/rancher/steve/pkg/auth/filter.go:172 +0x24f
net/http.HandlerFunc.ServeHTTP(0x50046e0?, {0x327b840?, 0xc006cf40e0?}, 0xc009d73b50?)
	/usr/local/go/src/net/http/server.go:2136 +0x29
net/http.serverHandler.ServeHTTP({0xc007ca7ce0?}, {0x327b840?, 0xc006cf40e0?}, 0x6?)
	/usr/local/go/src/net/http/server.go:2938 +0x8e
net/http.(*conn).serve(0xc00695e3f0, {0x328cdd0, 0xc0038a3560})
	/usr/local/go/src/net/http/server.go:2009 +0x5f4
created by net/http.(*Server).Serve in goroutine 1202
	/usr/local/go/src/net/http/server.go:3086 +0x5cb
```

**Related Issues:**
https://github.com/harvester/harvester/issues/158
https://github.com/harvester/harvester/issues/6105

**Test plan:**
- Hit https://192.168.0.131/v1/harvester/wtf and verify you get a 404, and the panic backtrace above is visible in the harvester pod logs
- It will be difficult to demonstrate a 500 internal server error, as I haven't found any other panics anywhere!

### CODE REVIEW

**Comment by tserong** [comment]
OK, I've added a second commit here, which is a better version IMO, as it:

1. Returns the error string in the response body, which I think better matches the spirit of the request in #158 
2. Uses structured logging
3. Adds an API endpoint that panics for manual testing (`/v1/harvester/dont-panic`)

It still has the side effect of "fixing" #6105 from the client's perspective.

**Comment by tserong** [comment]
Example log of hitting the test endpoint at `/v1/harvester/dont-panic`:

`time="2024-07-03T04:58:09Z" level=error msg="Recovered panic in Routes" err="Do you know where your towel is?" stack="goroutine 5787 [running]:\nruntime/debug.Stack()\n\t/usr/local/go/src/runtime/debug/stack.go:24 +0x5e\ngithub.com/harvester/harvester/pkg/server.recoveryHandler.ServeHTTP.func1()\n\t/go/src/github.com/harvester/harvester/pkg/server/router.go:133 +0x165\npanic({0x251b0a0?, 0x3221790?})\n\t/usr/local/go/src/runtime/panic.go:914 +0x21f\ngithub.com/harvester/harvester/pkg/server.(*Router).Routes.func2({0xc005159e30?, 0xc00bc49980?}, 0x10?)\n\t/go/src/github.com/harvester/harvester/pkg/server/router.go:66 +0x25\nnet/http.HandlerFunc.ServeHTTP(0xc005159e30?, {0x32402f8?, 0xc00e319260?}, 0x2b13160?)\n\t/usr/local/go/src/net/http/server.go:2141 +0x29\ngithub.com/harvester/harvester/vendor/github.com/rancher/apiserver/pkg/urlbuilder.RedirectRewrite.func1({0x32402f8?, 0xc00e319260}, 0xc00a4f7400)\n\t/go/src/github.com/harvester/harvester/vendor/github.com/rancher/apiserver/pkg/urlbuilder/redirect.go:14 +0x152\nnet/http.HandlerFunc.ServeHTTP(0xc00a4f7300?, {0x32402f8?, 0xc00e319260?}, 0x4187e8?)\n\t/usr/local/go/src/net/http/server.go:2141 +0x29\ngithub.com/harvester/harvester/vendor/github.com/gorilla/mux.(*Router).ServeHTTP(0xc0001e9d40, {0x32402f8, 0xc00e319260}, 0xc00a4f7200)\n\t/go/src/github.com/harvester/harvester/vendor/github.com/gorilla/mux/mux.go:212 +0x1c5\ngithub.com/harvester/harvester/pkg/server.recoveryHandler.ServeHTTP({{0x32259e0?, 0xc0001e9d40?}}, {0x32402f8?, 0xc00e319260?}, 0x1b95f01?)\n\t/go/src/github.com/harvester/harvester/pkg/server/router.go:138 +0x78\ngithub.com/harvester/harvester/vendor/github.com/rancher/steve/pkg/aggregation.ListenAndServe.ListenAndServe.ToMiddleware.func1.func2({0x32402f8, 0xc00e319260}, 0x0?)\n\t/go/src/github.com/harvester/harvester/vendor/github.com/rancher/steve/pkg/auth/filter.go:172 +0x24f\nnet/http.HandlerFunc.ServeHTTP(0x4f78fe0?, {0x32402f8?, 0xc00e319260?}, 0xc007e37b50?)\n\t/usr/local/go/src/net/http/server.go:2141 +0x29\nnet/http.serverHandler.ServeHTTP({0xc00d507d40?}, {0x32402f8?, 0xc00e319260?}, 0x6?)\n\t/usr/local/go/src/net/http/server.go:2943 +0x8e\nnet/http.(*conn).serve(0xc006925320, {0x32516c0, 0xc00342e180})\n\t/usr/local/go/src/net/http/server.go:2014 +0x5f4\ncreated by net/http.(*Server).Serve in goroutine 942\n\t/usr/local/go/src/net/http/server.go:3091 +0x5cb\n"`

It's all on one line, because of the structured logging, but the stack trace is easy enough to read if you just copy the quoted string out and paste it into a `printf` invocation in a shell, e.g.:

```
> printf ""goroutine 5787 [running]:\nruntime/debug.Stack()\n\t [...rest of string here...]
goroutine 5787 [running]:
runtime/debug.Stack()
	/usr/local/go/src/runtime/debug/stack.go:24 +0x5e
github.com/harvester/harvester/pkg/server.recoveryHandler.ServeHTTP.func1()
	/go/src/github.com/harvester/harvester/pkg/server/router.go:133 +0x165
panic({0x251b0a0?, 0x3221790?})
	/usr/local/go/src/runtime/panic.go:914 +0x21f
github.com/harvester/harvester/pkg/server.(*Router).Routes.func2({0xc005159e30?, 0xc00bc49980?}, 0x10?)
	/go/src/github.com/harvester/harvester/pkg/server/router.go:66 +0x25
net/http.HandlerFunc.ServeHTTP(0xc005159e30?, {0x32402f8?, 0xc00e319260?}, 0x2b13160?)
	/usr/local/go/src/net/http/server.go:2141 +0x29
github.com/harvester/harvester/vendor/github.com/rancher/apiserver/pkg/urlbuilder.RedirectRewrite.func1({0x32402f8?, 0xc00e319260}, 0xc00a4f7400)
	/go/src/github.com/harvester/harvester/vendor/github.com/rancher/apiserver/pkg/urlbuilder/redirect.go:14 +0x152
net/http.HandlerFunc.ServeHTTP(0xc00a4f7300?, {0x32402f8?, 0xc00e319260?}, 0x4187e8?)
	/usr/local/go/src/net/http/server.go:2141 +0x29
github.com/harvester/harvester/vendor/github.com/gorilla/mux.(*Router).ServeHTTP(0xc0001e9d40, {0x32402f8, 0xc00e319260}, 0xc00a4f7200)
	/go/src/github.com/harvester/harvester/vendor/github.com/gorilla/mux/mux.go:212 +0x1c5
github.com/harvester/harvester/pkg/server.recoveryHandler.ServeHTTP({{0x32259e0?, 0xc0001e9d40?}}, {0x32402f8?, 0xc00e319260?}, 0x1b95f01?)
	/go/src/github.com/harvester/harvester/pkg/server/router.go:138 +0x78
github.com/harvester/harvester/vendor/github.com/rancher/steve/pkg/aggregation.ListenAndServe.ListenAndServe.ToMiddleware.func1.func2({0x32402f8, 0xc00e319260}, 0x0?)
	/go/src/github.com/harvester/harvester/vendor/github.com/rancher/steve/pkg/auth/filter.go:172 +0x24f
net/http.HandlerFunc.ServeHTTP(0x4f78fe0?, {0x32402f8?, 0xc00e319260?}, 0xc007e37b50?)
	/usr/local/go/src/net/http/server.go:2141 +0x29
net/http.serverHandler.ServeHTTP({0xc00d507d40?}, {0x32402f8?, 0xc00e319260?}, 0x6?)
	/usr/local/go/src/net/http/server.go:2943 +0x8e
net/http.(*conn).serve(0xc006925320, {0x32516c0, 0xc00342e180})
	/usr/local/go/src/net/http/server.go:2014 +0x5f4
created by net/http.(*Server).Serve in goroutine 942
	/usr/local/go/src/net/http/server.go:3091 +0x5cb
```

**Review by Yu-Jack — COMMENTED** [review]
I think the main point is here, just add more information about apiserver.

https://github.com/rancher/apiserver/commit/8ad6ac8d69cd5b0605340047c7ea9e675544aac2#diff-104a11712684e520f0affa95e6a053faeb713306a10f69ee99b9729cc9dd1996

However, our `rancher/steve` isn't upgraded yet. Basically, `apiOp.Schema.ID` and `apiOp.Type` are the same when the request matches the schema. 

```sh
// logs from my harvester
apiOp.Schema.ID:  metrics.k8s.io.nodemetrics
apiOp.Type:  metrics.k8s.io.nodemetrics
```

In this commit, it replaced `apiOp.Schema.ID` with `apiOp.Type`. Because the request doesn't match schema, the `apiOp.Schema` is nil like you mentioned, but `apiOp.Type` will be the value we typed, in this example, would be `wtf`. If you'd like to try that, you could just modify the vendor codes to do that. Just one single line. 

```go
// from
metrics.RecordResponseTime(apiOp.Schema.ID, apiOp.Method, strconv.Itoa(code), float64(time.Since(requestStart).Milliseconds()))

// to
metrics.RecordResponseTime(apiOp.Type, apiOp.Method, strconv.Itoa(code), float64(time.Since(requestStart).Milliseconds()))
```

Then you'll get the 404. So, there are some choices we might have:

1. upgrade rancher/steve
2. keep recover implementation
3. modify vendor code

To be honest, I don't want third one. It'd be hard to manage.

**Comment by tserong** [comment]
> Then you'll get the 404. So, there are some choices we might have:
> 
>     1. upgrade rancher/steve
>     2. keep recover implementation
>     3. modify vendor code
> 
> To be honest, I don't want third one. It'd be hard to manage.

Yeah, I don't want the third one either :-)

My personal preference is to do both 1 (which I think really warrants a separate PR) _and_ 2 (this PR), because 2 gives us #158.

**Comment by Yu-Jack** [comment]
> 
> Yeah, I don't want the third one either :-)
> 
> My personal preference is to do both 1 (which I think really warrants a separate PR) _and_ 2 (this PR), because 2 gives us #158.

Agree that, we still need the recovery to prevent panic. 👍

**Review by Yu-Jack — COMMENTED** [review]
Maybe we could use `Middleware`? I'm thinking this might reduce some wrappers to make code look more straightforward. WDYT?

```go
m.Use(RecoverMiddleware)

func RecoverMiddleware(next http.Handler) http.Handler {
	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
		defer func() {
			if err := recover(); err != nil {
				w.WriteHeader(http.StatusInternalServerError)
				w.Write([]byte(fmt.Sprint(err)))
				logrus.WithFields(logrus.Fields{
					"err":   err,
					"stack": string(debug.Stack()),
				}).Error("Recovered panic in Routes")
			}
		}()
		next.ServeHTTP(w, r)
	})
}

```

**Comment by tserong** [comment]
> Maybe we could use `Middleware`? I'm thinking this might reduce some wrappers to make code look more straightforward. WDYT?

Done. I've also squashed this back to one commit with a simple explanation in the commit message of how this implements #158. I've left out the prior mention of how this also (kinda) fixes #6105 as a side-effect because as discussed earlier we're better off doing that by updating rancher/steve.

File: pkg/server/router.go [hunk 1]
```diff
@@ -59,6 +62,11 @@ func (r *Router) Routes(h router.Handlers) http.Handler {
 	m.Path("/v1/harvester/backuptarget/healthz").Methods("GET").Handler(btHealthyHandler)
 	// --- END of preposition routes ---
 
+	// This is for manually testing the recovery handler below
+	m.HandleFunc("/v1/harvester/dont-panic", func(_ http.ResponseWriter, _ *http.Request) {
+		panic("Do you know where your towel is?")
+	})
+
```
Comment (Yu-Jack) [inline pkg/server/router.go:69]: Do we need to remove this line if we're already tested it?

**Review by Yu-Jack — APPROVED** [review]
LGTM

File: pkg/server/router.go (hunk 1 above)
Comment (tserong) [reply pkg/server/router.go:69]: I wasn't sure about that. I tested it just now while updating this PR, so I know it works, and it is a pretty silly endpoint :-) OTOH, if we remove it, there's no way to manually test the recovery handler in future (think: regression testing).

**Review by bk201 — APPROVED** [review]
LGTM, test OK.

File: pkg/server/router.go (hunk 1 above)
Comment (Yu-Jack) [reply pkg/server/router.go:69]: I come up with a idea, we just need to write the mock server test case. I think it should work.

```go
m := mux.NewRouter()
m.Use(recoveryMiddleware)

// put your dont-panic here
m.HandleFunc("/v1/harvester/dont-panic", func(_ http.ResponseWriter, _ *http.Request) {
    panic("Do you know where your towel is?")
})

// new request
req, err := http.NewRequest("GET", "/v1/harvester/dont-panic", nil)

// the use httptest to mock server
responseWriter := httptest.NewRecorder()
m.ServeHTTP(responseWriter, req)

// then check responseWriter
```

Comment (tserong) [reply pkg/server/router.go:69]: AIUI that'll test the recoveryMiddleware itself works (good!) but won't provide any guarantee that the middleware is actually used in our router at runtime.

Comment (Yu-Jack) [reply pkg/server/router.go:69]: Okay, let's keep it!

Comment (tserong) [reply pkg/server/router.go:69]: OK, thanks for the review!

**Comment by tserong** [comment]
@mergifyio backport v1.4

**Pull request merged**